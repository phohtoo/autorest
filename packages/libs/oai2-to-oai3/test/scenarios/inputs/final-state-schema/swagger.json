{
  "swagger": "2.0",
  "info": {
    "version": "1.0",
    "title": "final-state-schema Test v1",
    "description": "This is a test of the final-state-schema",
    "contact": {
      "name": "John Doe",
      "email": "johndoe@contoso.com"
    }
  },
  "host": "contoso.com",
  "schemes": ["https"],
  "produces": ["application/json"],
  "tags": [
    {
      "name": "Products"
    }
  ],
  "paths": {
    "/products/{id}": {
      "put": {
        "operationId": "product_create",
        "description": "Create a Product",
        "tags": ["Products"],
        "x-ms-long-running-operation": true,
        "x-ms-long-running-operation-options": {
          "final-state-via": "location",
          "final-state-schema": "#/definitions/Product"
        },
        "parameters": [
          {
            "name": "id",
            "in": "path",
            "description": "Product ID",
            "required": true,
            "type": "string"
          },
          {
            "name": "body",
            "in": "body",
            "description": "Product object",
            "required": true,
            "schema": {
              "$ref": "#/definitions/Product"
            }
          }
        ],
        "responses": {
          "202": {
            "description": "Accepted"
          }
        }
      }
    }
  },
  "definitions": {
    "Product": {
      "properties": {
        "id": {
          "type": "string"
        }
      }
    }
  }
}
