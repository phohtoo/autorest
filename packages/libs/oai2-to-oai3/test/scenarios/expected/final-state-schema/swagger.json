{
  "servers": [
    {
      "url": "https://contoso.com/"
    }
  ],
  "openapi": "3.0.0",
  "info": {
    "version": "1.0",
    "title": "final-state-schema Test v1",
    "description": "This is a test of the final-state-schema",
    "contact": {
      "name": "John Doe",
      "email": "johndoe@contoso.com"
    }
  },
  "tags": [
    {
      "name": "Products"
    }
  ],
  "paths": {
    "/products/{id}": {
      "put": {
        "operationId": "product_create",
        "description": "Create a Product",
        "tags": [
          "Products"
        ],
        "x-ms-long-running-operation": true,
        "x-ms-long-running-operation-options": {
          "final-state-via": "location",
          "final-state-schema": "#/components/schemas/Product"
        },
        "parameters": [
          {
            "name": "id",
            "in": "path",
            "description": "Product ID",
            "required": true,
            "schema": {
              "type": "string"
            }
          }
        ],
        "requestBody": {
          "description": "Product object",
          "required": true,
          "x-ms-requestBody-name": "body",
          "content": {
            "application/json": {
              "schema": {
                "$ref": "#/components/schemas/Product"
              }
            }
          }
        },
        "responses": {
          "202": {
            "description": "Accepted"
          }
        }
      }
    }
  },
  "components": {
    "schemas": {
      "Product": {
        "properties": {
          "id": {
            "type": "string"
          }
        }
      }
    }
  }
}